### ISSUE TITLE
Status line attempts

### ISSUE THREAD

**Issue by lucabelmondo** [ygt-mikekchar/agnostic#9]
Hi @ygt-mikekchar 
I've been trying to find an alternative solution for the status line. It is very good in the dark variant, but I find it difficult in the light scheme.

Just to recap, this is the current version:

![screen shot 2015-04-01 at 09 25 38](https://cloud.githubusercontent.com/assets/698828/6937263/46babbdc-d852-11e4-9417-9e6f948cfa72.png)
![screen shot 2015-04-01 at 09 32 27](https://cloud.githubusercontent.com/assets/698828/6937264/46bc9d08-d852-11e4-8570-c39aa57721f4.png)

These are some combinations that I've tried, what do you think? It seems to be difficult to find something that has good readability in both variants.

![screen shot 2015-03-31 at 21 10 10](https://cloud.githubusercontent.com/assets/698828/6937295/8550c17a-d852-11e4-902f-d4b4be8eaebc.png)
![screen shot 2015-03-31 at 21 10 35](https://cloud.githubusercontent.com/assets/698828/6937292/854f141a-d852-11e4-9f78-619336cd3d34.png)
![screen shot 2015-04-01 at 09 23 20](https://cloud.githubusercontent.com/assets/698828/6937296/8551fb8a-d852-11e4-8b78-ad3e48b640ef.png)
![screen shot 2015-04-01 at 09 23 29](https://cloud.githubusercontent.com/assets/698828/6937294/854f98fe-d852-11e4-9e2a-bfc030530a98.png)
![screen shot 2015-04-01 at 09 23 57](https://cloud.githubusercontent.com/assets/698828/6937293/854f8aee-d852-11e4-9208-bca195a2d8bb.png)
![screen shot 2015-04-01 at 09 24 11](https://cloud.githubusercontent.com/assets/698828/6937297/8553fc3c-d852-11e4-8767-c70eb5df6488.png)
![screen shot 2015-04-01 at 09 24 59](https://cloud.githubusercontent.com/assets/698828/6937298/8562550c-d852-11e4-8989-8deca5b20df2.png)
![screen shot 2015-04-01 at 09 25 14](https://cloud.githubusercontent.com/assets/698828/6937299/85638c4c-d852-11e4-8d94-837dd50d988c.png)

**Comment by ygt-mikekchar** [comment]
Yes, I want to change that.  I also want to change the numberline background (the vertical bar with the line numbers in it).  I think that neither of them should be a normal foreground or background colour.  My initial though was to use colour 8 for the numberline background and unselected buffer modeline background and 7 for selected modeline background, but I have not had time to see what it looks like in all the supported palettes.

**Comment by lucabelmondo** [comment]
I've used this quick script to print all the color combinations: http://askubuntu.com/a/27318.

For the time being, just for the sake of readability, I would go with `ctermbg=3 ctermfg=0 cterm=none`. I will keep it in my config and if I find they work I will make a PR, or feel free to just add them to master yourself when you feel like it.

With regards to the styles per se, I would also like to address similar issues in the mode message and the error message. 

Looking at the bigger picture instead, while doing all the trials, I intuitively realised a couple of things, don't know if they make much sense, but here they go:

1 - most of the corresponding combinations in the two variants differ considerably in terms of contrast, so that most of good candidates in the light variant are not viable in the dark one instead; could it be that the bright colours in the dark variant are very dark compared to the background and to the normal colours? 

2 - within every variant, more so in the dark one, there are relatively few combinations with workable contrast, a part the ones including 0 and 8 in the light variant, and the 15 in the dark variant, but this limits the options because they can't be used at the same time;

I'm not sure they are actual issues and certainly don't know what to propose, but I will keep them in mind and will play with the hues to see if I can get any inspiration.

**Comment by ygt-mikekchar** [comment]
I think there are a couple of things going on:

1) The colours in the agnostic palettes were essentially chosen by me at
random to work well on that monitor I was using at work.  They may not
provide good contrast on other monitors.    I have some ideas for improving
the colours but haven't had time to try them out.
2) Remember that colours 0-6 are background colours and 9-15 are foreground
colours.  To help with Solarized compatibility, 7 and 8 are flipped (7 is a
foreground colour and 8 is a background colour).  You can never use 2
foreground colours together or 2 background colours together.  Similarly,
colours in the same hue are not guaranteed to have good contrast.  So you
can never use 1 with 9, 2 with 10, 3 with 11, etc.  Finally, complementary
colours will clash in many colour palettes (i.e. mixing red, green or blue
or mixing cyan, yellow or magenta).  This limits the available options
severely.

So basically you can mix background red, green, or blue with foreground
yellow, cyan, grey (7), or text colour(15).  You can mix background blue,
yellow or cyan with foreground red, green, blue, grey (7) or text colour
(15).  You can mix foreground text colour (15) with any background.  You
can mix background colour (0) with anything.  You can mix foreground
hilight grey (7) with any background colour.  You can mix background
highlight grey (8) without any foreground colour.

To further complicate things, some colour combinations just look awful in
the supported palettes.  I should point out what the supported palettes are
(I have a PR coming up with a better README and some up to date screen
shots).  The supported palettes are agnostic dark and light (of course),
linux console/xterm/iterm (they are all the same), solarized dark (but not
light), and tango (tango just happens to look great and is the default in
Ubuntu so I decided to give it supported status).  In an upcoming PR, I
will change the name of agnostic dark to agnostic-chalkboard and agnostic
light to agnostic-pastel.  I will add 2 new supported palettes that use
solarized colours.  These will be called agnostic-solarD an agnostic-solarL
(they use the exact colour values from solarized but have their values move
around so that agnostic looks closer to vim-colors-solarized with those
palettes).

All of the above should work and if they aren't visible with the agnostic
palettes, then it should be logged as a bug in the agnostic palettes.  We
should also have no other combinations in agnostic unless it is accompanied
by a comment explaining why it needs to be like that.

Hope that helps!  Thanks for this :-)

**Comment by mikekchar** [comment]
I've made considerable changes to the way borders are shown.  I cheekily merged into master :-P  When you get a chance, can you have a look and see if this issue is now resolved?
